README.md

```diff
@@ -1,50 +1,86 @@
 
-```bash
-make csi-provisioner
+## Compatibility
+
+This information reflects the head of this branch.
+
+| Compatible with CSI Version                                                                | Container Image                        | Min K8s Version |
+| ------------------------------------------------------------------------------------------ | ---------------------------------------| --------------- |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-provisioner:v1.0.1  | 1.13            |
+
+## Usage
+
+It is necessary to create a new service account and give it enough privileges to run the external-provisioner, see `deploy/kubernetes/rbac.yaml`. The provisioner is then deployed as single Deployment as illustrated below:
+
+```sh
+kubectl create deploy/kubernetes/deployment.yaml
 ```
 
-# Test
+The external-provisioner may run in the same pod with other external CSI controllers such as the external-attacher, external-snapshotter and/or external-resizer.
 
-## Start Kubernetes
+Note that the external-provisioner does not scale with more replicas. Only one external-provisioner is elected as leader and running. The others are waiting for the leader to die. They re-elect a new active leader in ~15 seconds after death of the old leader.
 
-Run a local kubernetes cluster built from latest master code
+### Command line options
 
-## Run Storage Provider
+#### Important optional arguments that are highly recommended to be used
+* `--csi-address <path to CSI socket>`: This is the path to the CSI driver socket inside the pod that the external-provisioner container will use to issue CSI operations (`/run/csi/socket` is used by default).
 
-### Use HostPath drivers
+* `--enable-leader-election`: Enables leader election. This is useful when there are multiple replicas of the same external-provisioner running for one CSI driver. Only one of them may be active (=leader). A new leader will be re-elected when current leader dies or becomes unresponsive for ~15 seconds.
 
-Go to drivers and run:
+* `--timeout <duration>`: Timeout of all calls to CSI driver. It should be set to value that accommodates majority of `ControllerCreateVolume` and `ControlerDeleteVolume` calls. See [CSI error and timeout handling](#csi-error-and-timeout-handling) for details. 15 seconds is used by default.
 
-```bash
-bin/hostpathplugin --drivername mydriver  --endpoint unix://tmp/csi.sock --nodeid foobar -v=5
-```
+* `--retry-interval-start <duration>` - Initial retry interval of failed provisioning or deletion. It doubles with each failure, up to `--retry-interval-max` and then it stops increasing. Default value is 1 second. [CSI error and timeout handling](#csi-error-and-timeout-handling) for details.
```

Comment:
external attacher doesn't have these options? should it?

Reply:
Attach/detach is done in clouds and it's "relatively" quick there. The options above are for the provisioners that take hours / days to create volume.